### PR TITLE
feat: derive tokenId from the ens domain

### PR DESCRIPTION
**Pull request by tiennampham23**
I see the [documetation](https://docs.ens.domains/dapp-developer-guide/ens-as-nft), the example is written by Javascript. So, I write the example with Go-ens

### CODE REVIEW

File: tokenid.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+package ens
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common/math"
+)
+
+// DeriveTokenIdFromENSDomain derive token_id from the ENS domain
+func DeriveTokenIdFromENSDomain(domain string) (string, error) {
```
Comment (mcdee) [inline tokenid.go:23]: I think this name can be shrunk to just `TokenID`; the rest is not adding a lot.

File: tokenid.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	domain, err := NormaliseDomain(domain)
+	if err != nil {
+		return "", err
+	}
+	domain, err = DomainPart(domain, 1)
+	if err != nil {
+		return "", err
+	}
+	labelHash, err := LabelHash(domain)
+	hash := fmt.Sprintf("0x%x", labelHash)
```
Comment (mcdee) [inline tokenid.go:33]: This format string could be written as `"%#x"`

File: tokenid.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+// limitations under the License.
+
+package ens
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common/math"
+)
+
+// DeriveTokenIdFromENSDomain derive token_id from the ENS domain
```
Comment (mcdee) [inline tokenid.go:22]: Would be good to have a bit more information about what the token ID is in the comment (specifically talk about it being the ERC-721 token ID).

File: tokenid_test.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package ens
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestDeriveTokenIdFromENSName(t *testing.T) {
```
Comment (mcdee) [inline tokenid_test.go:8]: Please could we add some tests that fail?  Empty domain, invalid domain, that type of thing.

**Review by mcdee — CHANGES_REQUESTED** [review]
Thank you for this submission.  I have requested a few alterations, but nothing major.

**Comment by tiennampham23** [comment]
Hi @mcdee, I resolved your comments, please review my PR again.

**Review by mcdee — APPROVED** [review]
Looks good, thanks for the changes.
